ctx_test.go

```diff
@@ -3059,6 +3067,40 @@ func Test_Ctx_QueryParser_Schema(t *testing.T) {
 	utils.AssertEqual(t, nil, c.QueryParser(n))
 	utils.AssertEqual(t, 3, n.Value)
 	utils.AssertEqual(t, 0, n.Next.Value)
+
+	type Person struct {
+		Name string `query:"name"`
+		Age  int    `query:"age"`
+	}
+
+	type CollectionQuery struct {
+		Data []Person `query:"data"`
+	}
+
+	c.Request().URI().SetQueryString("data[0][name]=john&data[0][age]=10&data[1][name]=doe&data[1][age]=12")
+	cq := new(CollectionQuery)
+	utils.AssertEqual(t, nil, c.QueryParser(cq))
+	utils.AssertEqual(t, 2, len(cq.Data))
+	utils.AssertEqual(t, "john", cq.Data[0].Name)
+	utils.AssertEqual(t, 10, cq.Data[0].Age)
+	utils.AssertEqual(t, "doe", cq.Data[1].Name)
+	utils.AssertEqual(t, 12, cq.Data[1].Age)
+
+	c.Request().URI().SetQueryString("data.0.name=john&data.0.age=10&data.1.name=doe&data.1.age=12")
+	cq = new(CollectionQuery)
+	utils.AssertEqual(t, nil, c.QueryParser(cq))
+	utils.AssertEqual(t, 2, len(cq.Data))
+	utils.AssertEqual(t, "john", cq.Data[0].Name)
+	utils.AssertEqual(t, 10, cq.Data[0].Age)
+	utils.AssertEqual(t, "doe", cq.Data[1].Name)
+	utils.AssertEqual(t, 12, cq.Data[1].Age)
+
+	c.Request().URI().SetQueryString("data[0]name]=john&data[0][age]=10&data[1][name]=doe&data[1][age]=12")
```

Comment:
Please add a "[" before name for the group 0.

Reply:
this is a negative test case to check if the `[` is ignored that the data will not be mapped

Reply:
ok I had not been able to see on my smartphone, then everything is okay